core/typedefs.h

```diff
@@ -357,4 +357,6 @@ struct _GlobalLock {
 #define FALLTHROUGH
 #endif
 
+#define MAX_RECURSION 1000
```

Comment:
GDScript defines `debug/settings/gdscript/max_call_stack [default: 1024]` in the project settings.

Reply:
That's a nice catch. However, I think it's better to keep them separate, since the GDScript max. stack size affects mainly the script realm, whereas `MAX_RECURSION` can be understood as a general limit for algorithms in the C++ realm that can potentially recurse very deeply. (Of course, the GDScript stack also has a correspondence in the C++ stack, but I'd still keep them as separate concerns,)